server/src/main/java/org/elasticsearch/common/lucene/search/function/ScriptScoreQuery.java

```diff
@@ -203,30 +213,33 @@ public int hashCode() {
     private static class ScriptScorer extends Scorer {
         private final ScoreScript scoreScript;
         private final Scorer subQueryScorer;
+        private final float boost;
         private final ExplanationHolder explanation;
 
         ScriptScorer(Weight weight, ScoreScript scoreScript, Scorer subQueryScorer,
-                ScoreMode subQueryScoreMode, ExplanationHolder explanation) {
+                ScoreMode subQueryScoreMode, float boost, ExplanationHolder explanation) {
             super(weight);
             this.scoreScript = scoreScript;
             if (subQueryScoreMode == ScoreMode.COMPLETE) {
                 scoreScript.setScorer(subQueryScorer);
             }
             this.subQueryScorer = subQueryScorer;
+            this.boost = boost;
             this.explanation = explanation;
         }
 
         @Override
         public float score() throws IOException {
             int docId = docID();
             scoreScript.setDocument(docId);
-            float score = (float) scoreScript.execute(explanation);
+            float score = (float) scoreScript.execute(explanation) * boost;
             if (score == Float.NEGATIVE_INFINITY || Float.isNaN(score)) {
                 throw new ElasticsearchException(
                     "script_score query returned an invalid score [" + score + "] for doc [" + docId + "].");
```

Comment:
I think it would be better if this error message returned the value produced by the script without the boost.